test/integration/lib/webpush_server.js

```diff
@@ -0,0 +1,93 @@
+'use strict';
+
+const express = require('express');
+const Config = require('config-js');
+const getRawBody = require('raw-body');
+const assert = require('assert');
+const crypto = require('crypto');
+const ece = require('http_ece');
+const typer = require('media-typer');
+var config = new Config('./test/integration/lib/config/foxbox.js');
+
+var webpush_server = (function() {
+  var serverECDH = crypto.createECDH('prime256v1');
+  var serverPublicKey = serverECDH.generateKeys('base64');
+  var _server = express();
+  
+  var port = config.get('webpush.port');
+  var endpoint = config.get('webpush.endpoint');
+  
+  var incomingPushMsg;
+  var serverInstance;
+
+  function getEndpointURI() {
+    return config.get('webpush.ip') + ':' + port + endpoint;
+  }
+
+  function getPublicKey() {
+    return serverPublicKey;
+  }
+
+  function getDecodedPushMsg() {
+    return incomingPushMsg;
+  }
+
+  function setup() {
+
+    //capture the payload of the message in a buffer
+    _server.use(function (req, res, next) {      
+      getRawBody(req, {
+        length: req.headers['content-length'],
+        limit: '1mb',
+        encoding: typer.parse('application/octet-stream').parameters.charset
+      }, function (err, data) {
+          if (err) {return next(err);}
+          req.data = data;
+          next();
+        });
+    });
+  
+    // Handles the incoming ECDH encrypted webpush message from foxbox
+    _server.post(endpoint, function (req, res) {
+      assert.equal(req.headers['content-encoding'], 
+        'aesgcm128', 'Content-Encoding header correct');
+
+      // Collect salt, publickey, and the message
+      var saltcut = req.headers.encryption.replace(/keyid=p256dh;salt=/g,'');
+      var pkcut = req.headers['encryption-key'].replace(/keyid=p256dh;dh=/g,'');
+      var encryptedBuffer = new Buffer(req.data,'binary');
+      
+      // compute shared secret using the provided public key of foxbox
+      var sharedSecret = serverECDH.computeSecret(pkcut,'base64');
+      ece.saveKey('webpushKey', sharedSecret);
+        
+      // Decrypt the message using the shared secret and provided salt
+      var decrypted = ece.decrypt(encryptedBuffer, {
+              keyid: 'webpushKey',
+              salt: saltcut,
+              padSize: 1,
+            });
+      incomingPushMsg = decrypted.toString();        
```

Comment:
This looks okay for now, but we might face race conditions if many responses are received almost at the same time. Maybe `incomingPushMsg` could be an array?